src/shared/components/app/app.tsx

```diff
@@ -29,6 +29,7 @@ export class App extends Component<AppProps, any> {
             <Theme
               myUserInfo={siteRes.my_user}
               defaultTheme={siteRes?.site_view?.site?.default_theme}
+              defaultCodeTheme={siteRes?.site_view?.site?.default_code_theme}
```

Comment:
Lint is failing because this field doesn't exist in the back end.

Rather than have this be customizeable in the back end, why not just do one of the following:

- Have a default light and dark theme, and maybe default to dark in other cases.
- Possibly create a javascript map from theme to hljs theme ( in src/server/index.tsx ) , although this seems overkill to me.

Reply:
If were to have a 1->1 relation between themes and code-themes, I thought my previous solution was better because 1. mainline theme creators could pick the most appropriate theme, and 2. custom themes can include an appropriate theme. 

Another possibility would be to measure the human-perceived luminosity of the themes background color and pick between light and dark code-themes. I've done this before and it worked well, but I would prefer to have user-selectable themes =/

Reply:
I explained here why i dont think it makes sense to add a setting for this:

https://github.com/LemmyNet/lemmy/issues/2307#issuecomment-1151615448

Reply:
I think my first proposal makes sense: pick a good light and dark theme, and use `prefers-color-scheme` to decide the right one to load.

Reply:
If people set a UI-theme that conflicts with their browser theme the only solution is to change their browser theme.  Any sites that set a default theme will have decoherent themes for a lot of users  =\

Additionally, there is no customizability and themers have their hands tied. I think importing hljs themes from UI-themes is more extensible, and doesn't pose a similar usability issue.

Reply:
I think the initial implementation with `@import` made most sense, because it allows each theme to explicitly specify a matching code theme. Plus if there is none that matches, parts of the code theme could be overwritten, or an entirely new theme included. `prefers-color-scheme` is the same thing thats still broken in Firefox, which makes Lemmy always use the light theme, right? I would hate to rely on something thats still completely broken for many users.

Reply:
Yep some browsers don't fully support it yet, I thought firefox did tho, at least FF mobile does. I was thinking the default for code would be dark. The @import s on each of the themes could work for a short term fix too tho.

Ideally, lemmy-ui would not store any minified themes, and only work with bootstrap `variables.scss` files, then generate the themes itself. That way we could get rid of them from the codebase. But its not necessary for this issue.